src/platform/f4xx/stm32f446xx.cpp

```diff
@@ -0,0 +1,75 @@
+#include <HALf4/stm32f4xx.h>
+#include <HALf4/stm32f4xx_hal.h>
+#include <HALf4/stm32f4xx_it.h>
+
+#include <EVT/platform/f4xx/stm32f4xx.hpp>
+#include <iostream>
+
+using namespace std;
+namespace EVT::core::platform {
+
+void stm32f4xx_init() {
+    HAL_Init();
+    RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+    RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+    RCC_PeriphCLKInitTypeDef PeriphClkInit = {0};
+
+    /** Configure the main internal regulator output voltage
+    */
+    __HAL_RCC_PWR_CLK_ENABLE();
+    __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE3);
+
+    /** Initializes the RCC Oscillators according to the specified parameters
+    * in the RCC_OscInitTypeDef structure.
+    */
+    RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSI;
+    RCC_OscInitStruct.HSIState = RCC_HSI_ON;
+    RCC_OscInitStruct.HSICalibrationValue = RCC_HSICALIBRATION_DEFAULT;
+    RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
+    RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_HSI;
+    RCC_OscInitStruct.PLL.PLLM = 8;
+    RCC_OscInitStruct.PLL.PLLN = 50;
+    RCC_OscInitStruct.PLL.PLLP = RCC_PLLP_DIV2;
+    RCC_OscInitStruct.PLL.PLLQ = 2;
+    RCC_OscInitStruct.PLL.PLLR = 2;
+    HAL_RCC_OscConfig(&RCC_OscInitStruct);
```

Comment:
Why is this different than the 3xx stuff?

Reply:
bc there are different scalars for the pll clock on the f4's than the f3's